README.md

```diff
@@ -109,6 +88,36 @@ Please also visit the `get_started` directory for an example configuration for p
 
 If you wish to work on the provider, you'll first need [Go](http://www.golang.org) installed on your machine (see [Requirements](#requirements) above). You will also need to have access to a [Couchbase Capella](https://www.couchbase.com/products/capella) account.
 
+## Building The Provider
+
+1. Clone the repository
+1. Enter the repository directory
+1. Build the provider using the Go `build` command:
+
+```sh
+$ go build
+```
+
+After the provider has been built locally it must be placed in the user plugins directory so it can be discovered by the Terraform CLI. Please execute the following command to move the provider binary to this directory:
+
+```sh
+$ mv terraform-provider-couchbasecapella ~/.terraform.d/plugins/local/couchbasecapella/<VERSION>/<OS_ARCH>
```

Comment:
I don't think using `local` works here:

```sh
» ls -1 ~/.terraform.d/plugins/local/couchbasecapella/1.0.0/
darwin_amd64

» cat versions.tf
terraform {
  required_providers {
    couchbasecapella = {
      source  = "local/couchbasecapella"
      version = "1.0.0"
    }
  }
}

» terraform init

Initializing the backend...

Initializing provider plugins...
- Finding local/couchbasecapella versions matching "1.0.0"...
╷
│ Error: Failed to query available provider packages
│
│ Could not retrieve the list of available versions for provider local/couchbasecapella: provider registry registry.terraform.io does not have a provider named
│ registry.terraform.io/local/couchbasecapella
```

IIRC the hostname when building locally is _kind of required_. If you omit it, it will use [the default registry](https://learn.hashicorp.com/tutorials/terraform/provider-use?in=terraform/providers#install-hashicups-provider) and it fails:

> If hostname is omitted, Terraform will use the Terraform Registry hostname as the default hostname.

Usually setting this to `github.com` is a good practice, as it matches the hostname/namespace from the repository on GitHub.

Also, the folder needs to be created first, maybe we can set up in another PR a target on the `Makefile` to automate this and improve the development flow (I can help with this)?

Reply:
Yeah I agree, that would be much better in the `Makefile`. We can just run the one command then. I'll create a branch off of this one and we can make a PR for Makefile/changes to docs